src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/ProxyRoutes.scala

```diff
@@ -1,48 +1,110 @@
 package org.broadinstitute.dsde.workbench.leonardo.api
 
+import akka.http.scaladsl.model.StatusCodes
 import akka.http.scaladsl.server.Directives._
-import akka.http.scaladsl.server.Route
+import akka.http.scaladsl.server.{AuthorizationFailedRejection, Directive0, Directive1, Route}
 import com.typesafe.scalalogging.LazyLogging
 import org.broadinstitute.dsde.workbench.leonardo.model.ClusterName
 import org.broadinstitute.dsde.workbench.leonardo.service.ProxyService
-import org.broadinstitute.dsde.workbench.model.WorkbenchEmail
 import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import akka.http.scaladsl.model.headers.{Authorization, HttpCookie}
+import org.broadinstitute.dsde.workbench.model.UserInfo
 
 import scala.concurrent.ExecutionContext
 
-/**
-  * Created by rtitle on 8/4/17.
-  */
-trait ProxyRoutes extends UserInfoDirectives{ self: LazyLogging =>
+trait ProxyRoutes extends UserInfoDirectives { self: LazyLogging =>
   val proxyService: ProxyService
   implicit val executionContext: ExecutionContext
 
   protected val tokenCookieName = "FCtoken"
 
   protected val proxyRoutes: Route =
+    path("notebooks" / "login") {
```

Comment:
All `login` really does is migrate a token from a header to a cookie. Is login the right word?

Also (I'm just thinking this): should this endpoint take a `googleProject/clusterName` and consult the authorization provider?

Reply:
`setCookie` seems a reasonable name to me.

If you don't consult the auth provider here, they'll get a 401/404 from it when they do either of the proxied actions. It doesn't seem super necessary to check it here, but I have no strong feelings either way.

Reply:
At some point the auth provider -- or some other pluggable -- is going to have to poke around in the header looking for the appropriate auth-y things, which might change depending on auth mechanism. No point demanding a `UserInfo` if Leo is being stood up unsecured in a lab somewhere.

Reply:
I'll rename to `setCookie`.

I think it would be nicer for the user to do the auth check here. This call is being done by AJAX _before_ opening the notebook in a new tab. If we did the auth check sooner rather than later it would give the UI more flexibility to deal with 401s before opening the new tab.